File: v-lang/l-java/v-utils/src/main/java/org/example/json/JacksonUtils.java
package org.example.json;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;

@UtilityClass
public class JacksonUtils {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @SneakyThrows
    public <T> T  json2Object(String json, Class<T> clazz) {
        return objectMapper.readValue(json, clazz);
    }

    @SneakyThrows
    public <T> T json2Object(String json, TypeReference<T> typeRef) {
        return objectMapper.readValue(json, typeRef);
    }

    @SneakyThrows
    public String object2Json(Object obj) {
        return objectMapper.writeValueAsString(obj);
    }
}

File: v-lang/kt.d/v-utils/u-json/src/main/kotlin/utils/JacksonUtils.kt
package org.example.utils

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper

object JacksonUtils {

    private val objectMapper = ObjectMapper()

    fun <T> json2Object(json: String, clazz: Class<T>): T = objectMapper.readValue(json, clazz)

    fun <T> json2Object(json: String, typeRef: TypeReference<T>): T = objectMapper.readValue(json, typeRef)

    fun object2Json(obj: Any): String = objectMapper.writeValueAsString(obj)

}